blackhole.c

```diff
@@ -601,13 +606,15 @@ blackhole_feedback_ngbiter(TreeWalkQueryBHFeedback * I,
 
         lock_particle(other);
 
+        /* Mass Conservation */
         O->Mass += (P[other].Mass);
+        P[other].Mass = 0;
```

Comment:
Ditto. Why zero the mass at all when you have IsGarbage?

Reply:
See the comment above. It signifies mass must be conserved. I keep this otherwise useless code there to remind you something potentially bad is going on -- We do not have a mechanism to notify the tree particles have turned into garbage or the mass of a particle has changed. And these operations, even emitted, shall be somehow translated into a noop, because we know the mass of the node is conserved. I haven't thought clear how to do this.

Reply:
Ah, I see. That makes a lot of sense, but still you need this to be set later, because otherwise AccretedMomentum would be zero.

Reply:
Yes I fixed that and added a rebase. Actually there is no guarentee the swallowee and the swallower are in the same tree node. So it won't always become a noop. I guess we need a function to 'transfer mass'.

Reply:
I can add that in a new PR. It could be gradually paving the road towards incremental tree.

Reply:
A very good point